mathics/builtin/comparison.py

```diff
@@ -219,75 +219,66 @@ def numerify_args(items, evaluation):
-        # See comments in
-        # [https://github.com/mathics/Mathics/pull/1209#issuecomment-810277502]
-        # for a future refactory of this methods...
-        if hasattr(lhs, "equal2"):
-            result = lhs.equal2(rhs)
-            if result is not None:
-                return result
-
-        # Do this after lhs.equal2 since lhs.equal2 will do a sameQ test.
-        if lhs.sameQ(rhs):
-            return True
-
-        # FIXME: I think this can be folded into Symbol.equal2 and Atom.equal2
-        if not (isinstance(lhs, Symbol) or isinstance(rhs, Symbol)) and (
-            isinstance(lhs, Atom) and isinstance(rhs, Atom)
-        ):
-            return lhs == rhs
-
-        # Dealing with two non-atomic expressions:
-        if not rhs.is_atom():
-            head1 = lhs.get_head()
-            head2 = rhs.get_head()
-
-            # FIXME: add equal2 method for DirectedInfinity
-
-            # Handling comparisons with DirectedInfinity
-            if head2.sameQ(SymbolDirectedInfinity):
-                lhs, rhs = rhs, lhs
-                head1, head2 = head2, head1
-            if head1.sameQ(SymbolDirectedInfinity):
-                if head2.sameQ(SymbolDirectedInfinity):
-                    dir1 = dir2 = Integer(1)
-                    if len(lhs._leaves) == 0:
-                        if len(rhs._leaves) == 0:
-                            return True
-                        dir1 = Integer(1)
-                    else:
-                        dir1 = lhs._leaves[0]
-                    if len(rhs._leaves) == 0:
-                        if dir1.sameQ(Integer(1)):
-                            return True
-                        dir2 = Integer(1)
-                    else:
-                        dir2 = rhs._leaves[0]
-                    # If the directions are equal,
-                    # then both infinites are the same
-                    if self.equal2(dir1, dir2):
-                        return True
-                    # Now, compare the signs:
-                    dir1 = Expression("Sign", dir1)
-                    dir2 = Expression("Sign", dir2)
-                    return self.equal2(dir1, dir2)
-
-        # Dealing with one expression
-        elif not lhs.is_atom():
-            # FIXME: Ass mentioned above: add equal2 method for DirectedInfinity
-            if lhs.get_head().sameQ(SymbolDirectedInfinity):
-                if isinstance(rhs, Number):
-                    return False
-                elif SymbolInfinity.sameQ(rhs):
-                    if len(lhs._leaves) == 0 or self.equal2(lhs._leaves[0], Integer(1)):
-                        return True
-
-        lhs_sympy = lhs.to_sympy(evaluate=True, prec=COMPARE_PREC)
-        rhs_sympy = rhs.to_sympy(evaluate=True, prec=COMPARE_PREC)
+    @staticmethod
+    def get_pairs(args):
+        for i in range(len(args)):
+            for j in range(i):
+                yield (args[i], args[j])
+
+    def expr_cmp(self, lhs, rhs, max_extra_prec=None):
```

Comment:
I would appreciate it if you not use the word `cmp` for `Equal` like functions. 

In Python (and other languages) `cmp`  returns some sort of integer where 0 means the items are equal and 1 (or more generally a positive number) means lhs > rhs and -1 (or more generally a negative number) means that lhs < rhs. 

And even in this code before this change, this convention is followed. 

Similarly WL took the name Equal from Lisp and used in other symbolic languages where that means about the same thing. WL uses SameQ for what in Lisp is `eq`. 

But what I am trying to get across here is that although this seems like a niggling concern, keeping things consistent is very helpful for someone steeping in programming languages and symbolic programming languages in particular and the conventions used there. 

So `expr_equal` might be a possibility because it remove `cmp` which shouldn't be used and adds `equal` which should be used. However the exact name I don't care about as long as it doesn't go against existing conventions. 

Also, it would be helpful to me if you'd add the return value `-> Optional[bool]` (which is the same thing as `Union[bool, None]` but more clear. I am sorry I didn't do this initially.